### DISCUSSION TITLE
Request headers are not passed to the play.http.actionCreator class post 2.8 upgrade

### DISCUSSION THREAD

**Discussion by ilapavuluri** [playframework/playframework#11880] [Q&A] [answered]
Hi Team,

We are using play framework which has custom filters which extends EssentialFilter and a play.http.actionCreator implementation.

For Authentication it first hits custom filter which extends EssentialFilter and later our play.http.actionCreator implementation.

this was working fine with older play versions, once after we upgrade to play 2.8, it is not carry forwarding headers automatically to 
play.http.actionCreator implementation class from the filter class.

Please help us to resolve , this became a blocker for our play upgrade.

**Comment by mkurz** [comment]
You mean you set headers in the filter? Can you please give some code examples what you do exactly?

**Comment by ilapavuluri** [comment]
@mkurz Thank you for your help.

In one of our filters (which extends EssentialFilter) we are setting a custom header like this:
.withHeader(CUSTOM_HEADER, value), this is not persisting in the subsequent calls...example: after executing this filter it comes to our AuthenticationInterceptor (which implements ActionCreator), here I am not seeing those custom headers.


If it is a stranded header like .withHeader(HEADER_CONTENT_SECURITY_POLICY, cspFilterValue); it is working fine.

We uses Symantec SiteMinder for SSO authentication, whatever custom headers send by SiteMinder I am able to see in the Filter(s) class but they are not coming to the ActionCreator class.

This was working fine in the play 2.3 and started failing after upgrade to 2.8 (since we don't have context now)

This became a blocker for our release, appreciate your help.

Thanks
Ashok.

**Comment by ilapavuluri** [comment]
@mkurz any inputs on the above? appreciate your help.

Thanks
Ashok.

**Comment by gmethvin** [comment, marked as answer]
> In one of our filters (which extends EssentialFilter) we are setting a custom header like this:
.withHeader(CUSTOM_HEADER, value), this is not persisting in the subsequent calls...example: after executing this filter it comes to our AuthenticationInterceptor (which implements ActionCreator), here I am not seeing those custom headers.

You're referring to the `withHeader` method on the `Result`? When you return a result from the filter, your filter would have already called the action. I'm not sure where you're expecting to see the header.

> This was working fine in the play 2.3 and started failing after upgrade to 2.8 (since we don't have context now)

OK, I think I'm understanding... You were originally on Play 2.3, and you were setting a _response_ header using the request's `Http.Context` instance, then retrieving that from the context in a later action in the chain? If so, then there's no way to do the same thing in Play anymore, but you might be able to achieve the same effect by attaching the contextual information in [request attributes](https://www.playframework.com/documentation/2.8.x/Highlights26#Request-attributes).

**Comment by gmethvin** [comment]
It sounds like this is an issue with your migration rather than a bug in Play, so I converted this to a discussion.

**Comment by ilapavuluri** [comment]
@gmethvin  Thank you so much for the help! I completely agree this is an issue with our migration, will try the suggestions provided by you and keep you posted.

Once again thank you.

Thanks
Ashok.

**Comment by ilapavuluri** [comment]
@gmethvin thank you so much for your inputs, the above suggestions really helped and resolved my problem, thanks again.

**Reply by gmethvin** [reply]
You're welcome, glad you got it working!
